src/AutoRest.CSharp/Common/Input/CadlInputConstantConverter.cs

```diff
@@ -93,6 +93,22 @@ public static object ReadConstantValue(ref Utf8JsonReader reader, string propert
 
                     }
                     break;
+                case InputEnumType enumType:
+                    switch (enumType.EnumValueType.Kind)
+                    {
+                        case InputTypeKind.String:
+                            value = reader.GetString() ?? throw new JsonException();
+                            break;
+                        case InputTypeKind.Int32:
+                            value = reader.GetInt32();
+                            break;
```

Comment:
Just curious, this is not related to the APIversion, right?

Reply:
Yes, It is not for apiVersion. this is to support  InputConstant for enum type.